core/src/code.rs

```diff
@@ -219,36 +221,95 @@ impl Code {
 
         let exports = get_exports(&module, false)?;
 
-        if exports.contains(&DispatchKind::Init) || exports.contains(&DispatchKind::Handle) {
-            if instrument_with_const_rules {
-                let instrumented_module =
-                    gear_wasm_instrument::inject(module, &ConstantCostRules::default(), "env")
-                        .map_err(|_| CodeError::GasInjection)?;
-
-                let instrumented = parity_wasm::elements::serialize(instrumented_module)
-                    .map_err(|_| CodeError::Encode)?;
-
-                Ok(Self {
-                    raw_code: original_code,
-                    code: instrumented,
-                    exports,
-                    static_pages,
-                    instruction_weights_version: version,
-                })
-            } else {
-                Ok(Self {
-                    raw_code: original_code.clone(),
-                    code: original_code,
-                    exports,
-                    static_pages,
-                    instruction_weights_version: version,
-                })
-            }
+        if check_entries
+            && !(exports.contains(&DispatchKind::Init) || exports.contains(&DispatchKind::Handle))
+        {
+            return Err(CodeError::RequiredExportFnNotFound);
+        }
+
+        if instrument_with_const_rules {
+            let instrumented_module =
+                gear_wasm_instrument::inject(module, &ConstantCostRules::default(), "env")
+                    .map_err(|_| CodeError::GasInjection)?;
+
+            let instrumented = parity_wasm::elements::serialize(instrumented_module)
+                .map_err(|_| CodeError::Encode)?;
+
+            Ok(Self {
+                raw_code: original_code,
+                code: instrumented,
+                exports,
+                static_pages,
+                instruction_weights_version: version,
+            })
         } else {
-            Err(CodeError::RequiredExportFnNotFound)
+            Ok(Self {
+                raw_code: original_code.clone(),
+                code: original_code,
+                exports,
+                static_pages,
+                instruction_weights_version: version,
+            })
         }
     }
 
+    /// Create the code with instrumentation, but without checks.
+    /// There is also no check for static memory pages.
+    pub fn new_raw_with_rules<R, GetRulesFn>(
```

Comment:
Testing required

Reply:
Seems `new_raw` doesn't have tests also, but both of them are only for non-production solution: tests and benches